pkg/image/trigger/buildconfigs/buildconfigs.go

```diff
@@ -175,9 +176,22 @@ func (r *buildConfigReactor) ImageChanged(obj runtime.Object, tagRetriever trigg
 			newSource = true
 		}
 
-		// LastTriggeredImageID is an image ref, despite the name
-		if latest == p.LastTriggeredImageID {
-			continue
+		// see if this trigger has fired previously and is recorded in status;
+		// openshfit-apiserver is in charge of populating the status for image change triggers,
+		// including resolving LastTriggeredImageID and LastTriggerTime
+		pStatus := ocmbuildutil.GetImageChageTriggerStatusForImageChangeTrigger(p, bc)
+		if pStatus != nil {
+			// LastTriggeredImageID is an image ref, despite the name
+			if latest == pStatus.LastTriggeredImageID {
+				continue
+			}
+		} else {
+			//TODO until openshift-apiserver is updated to maintain image triggers in the status,
```

Comment:
Still a to-do item?

Reply:
yeah I have a commit locally that removes this, but I was holding out on pushing it up until https://github.com/openshift/openshift-apiserver/pull/191 merged @rolfedh 

but if I'm going to pull in your suggested changes for the comments, perhaps I'll just pull the trigger on this too, and we can live with any short term e2e failures until the apiserver PR merges